pyproject.toml

```diff
@@ -3,7 +3,7 @@
 requires = [
     "setuptools >=61.0.0",
     "wheel",
-    "oldest-supported-numpy",
+    "numpy", #"oldest-supported-numpy",
```

Comment:
Just for curiosity, is there any specific issue with "oldest-supported-numpy"?

Reply:
Hi @avalentino, commit fba55dd45 message indicates "This should avoid requiring very old versions of Blosc2 for older Python 3.8 and 3.9.", but @FrancescAlted's commit e6086fa06 indicates "Fallback to use oldest-supported-numpy (blosc2 should support this now)". I see that Blosc2 depends on oldest-supported-numpy from v0.6.3.

I recall having discussed this with @FrancescAlted, he can probably offer a better explanation for the change (that I should definitely provide in a comment or in the commit message :sweat_smile: ).

Thanks for having a look!

Reply:
Yeah, so sorry, but I don't remember the details, but IIRC during our recent work, we detected that  `oldest-supported-numpy` gave problems, so I decided to give regular "numpy" a try, and that fixed the issue. 

Anyway, by looking at https://pypi.org/project/oldest-supported-numpy/2023.8.3, building with that makes sense to me.  So I'd suggest to enforce `oldest-supported-numpy` again and retry :-)

Reply:
I agree

Reply:
While working on the branch for the v3.9.0 release, commit 626de54d did update Blosc2 dependencies, and this resulted in CI errors like <https://github.com/PyTables/PyTables/actions/runs/6248015264/job/16961771859#step:6:100>, which may clarify the issue:

```
ERROR: Cannot install blosc2==2.2.6, blosc2==2.2.7 and oldest-supported-numpy==0.1 because these package versions have conflicting dependencies.

  The conflict is caused by:
      oldest-supported-numpy 0.1 depends on numpy==1.17.3; python_version >= "3.8" and platform_system != "AIX"
      blosc2 2.2.7 depends on numpy>=1.20.3
      oldest-supported-numpy 0.1 depends on numpy==1.17.3; python_version >= "3.8" and platform_system != "AIX"
      blosc2 2.2.6 depends on numpy>=1.20.3
```

I then cherry-picked fba55dd for that branch, added commit a203326 with a better commit message, and CI seems to be happy with that. So I may revert the change in this PR's branch, but make the change in the release branch. Or I can undo the changes in the latter too if they don't look like the way to go yet. :wink:

Reply:
Thanks @ivilata for taking care of this.
In this case, maybe we should update the GHS for building wheel to use `numpy==1.20.3`.

Reply:
You're welcome @avalentino! After commit a203326, in the wheels CI action the "Build source/Install dependencies" decides to install NumPy 1.26.0 and everything seems to work fine: <https://github.com/PyTables/PyTables/actions/runs/6261949618>. Do you think we really should set a fixed version, or is it ok to just let `pip` et al. decide on their own?